Marlin/src/lcd/extensible_ui/ui_api.cpp

```diff
@@ -577,6 +577,27 @@ namespace ExtUI {
     return elapsed.value;
   }
 
+   #if HAS_LEVELING
+    bool getLevelingActive() { return planner.leveling_active; }
+    void setLevelingActive(const bool state) { set_bed_leveling_enabled(state) }
+    #if HAS_MESH
+      bool getMeshValid() { return leveling_is_valid(); }
+      float[GRID_MAX_POINTS_X][GRID_MAX_POINTS_Y] getMeshArray() { return Z_VALUES; }
```

Comment:
What's this? I've never seen anyone return an array from a function. Would you just get a pointer out of this? Isn't the dimensions info meaningless?

Reply:
I believe C++ will actually type-check this if you're assigning to an array type. The details can be concealed a bit by defining a type for the array:
```cpp
typedef float[GRID_MAX_POINTS_X][GRID_MAX_POINTS_Y] bed_mesh_t;
```

Reply:
Nope! C++ considers this gibberish.

Reply:
Correct approach is to use an array reference…
```cpp
float (&getMeshArray())[GRID_MAX_POINTS_X][GRID_MAX_POINTS_Y] {
  return Z_VALUES;
}
```
…or…
```cpp
typedef float (&bed_mesh_t)[GRID_MAX_POINTS_X][GRID_MAX_POINTS_Y];
bed_mesh_t getMeshArray() {  return Z_VALUES; }
```

Reply:
Whoops, probably pasted it while in auto pilot not thinking.... Ran through stuff today while installing software updates for the day job so wasn't really paying close enough attention.